packages/app/src/app/hooks/useIsFirstVisit.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffects } from 'app/overmind';
+import { useEffect } from 'react';
+
+const KEY = 'DASHBOARD_FIRST_VISIT';
+
+export const useIsFirstVisit = () => {
+  const { browser } = useEffects();
+
+  const isFirstVisit = browser.storage.get<boolean>(KEY) ?? true;
+
+  useEffect(() => {
+    const setHasVisited = () => {
+      browser.storage.set(KEY, false);
+    };
+
+    window.addEventListener('beforeunload', setHasVisited);
```

Comment:
I'm not sure if the `beforeunload` listener is the best option here. I've been reading about it a bit and both [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) and [Chrome](https://developer.chrome.com/blog/page-lifecycle-api/) don't recommend it for cases like these. The code now only triggers when the hook is rendered on a page, so when moving from `/recent` to `/discover` and then closing the website does not trigger `setHasVisited`.

I'm not sure what we define as a "first visit", is it actually closing CodeSandbox after the first time, or is it also moving to a different page than recent? This does make implementation harder. The closest thing tracking the first visit - no matter which page on the dashboard has been visited - would be to actually check the second visit.